cxx-qt-lib/src/qdatetime.rs

```diff
@@ -0,0 +1,198 @@
+// on the stack in pinned form which avoids the pitfalls of trying to
+// move this type that has a non-trivial move constructor.
+unsafe impl ExternType for QDateTime {
+    type Id = type_id!("QDateTime");
+    type Kind = cxx::kind::Opaque;
+}
+
+extern "C" {
+    #[link_name = "cxxqt1$unique_ptr$qdatetime$null"]
+    fn unique_ptr_qdatetime_null(this: *mut MaybeUninit<*mut c_void>);
+    #[link_name = "cxxqt1$unique_ptr$qdatetime$raw"]
+    fn unique_ptr_qdatetime_raw(this: *mut MaybeUninit<*mut c_void>, raw: *mut QDateTime);
+    #[link_name = "cxxqt1$unique_ptr$qdatetime$get"]
+    fn unique_ptr_qdatetime_get(this: *const MaybeUninit<*mut c_void>) -> *const QDateTime;
+    #[link_name = "cxxqt1$unique_ptr$qdatetime$release"]
+    fn unique_ptr_qdatetime_release(this: *mut MaybeUninit<*mut c_void>) -> *mut QDateTime;
+    #[link_name = "cxxqt1$unique_ptr$qdatetime$drop"]
+    fn unique_ptr_qdatetime_drop(this: *mut MaybeUninit<*mut c_void>);
+}
+
+// Safety: TODO
+unsafe impl UniquePtrTarget for QDateTime {
+    #[doc(hidden)]
+    fn __typename(f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        f.write_str("QDateTime")
+    }
+
+    #[doc(hidden)]
+    fn __null() -> MaybeUninit<*mut c_void> {
+        let mut repr = MaybeUninit::uninit();
+        unsafe {
+            unique_ptr_qdatetime_null(&mut repr);
+        }
+        repr
+    }
+
+    #[doc(hidden)]
+    unsafe fn __raw(raw: *mut Self) -> MaybeUninit<*mut c_void> {
+        let mut repr = MaybeUninit::uninit();
+        unique_ptr_qdatetime_raw(&mut repr, raw);
+        repr
+    }
+
+    #[doc(hidden)]
+    unsafe fn __get(repr: MaybeUninit<*mut c_void>) -> *const Self {
+        unique_ptr_qdatetime_get(&repr)
+    }
+
+    #[doc(hidden)]
+    unsafe fn __release(mut repr: MaybeUninit<*mut c_void>) -> *mut Self {
+        unique_ptr_qdatetime_release(&mut repr)
+    }
+
+    #[doc(hidden)]
+    unsafe fn __drop(mut repr: MaybeUninit<*mut c_void>) {
+        unique_ptr_qdatetime_drop(&mut repr)
+    }
+}
+
+/// The Rust representation of Qt's QDateTime
+///
+/// Internally this holds a UniquePtr to a QDateTime which has been constructed on the C++ side.
+pub struct DateTime {
+    // Note that once map_qt_value is removed later, this can become private again
+    #[doc(hidden)]
+    pub(crate) inner: cxx::UniquePtr<QDateTime>,
+}
+
+impl DateTime {
+    /// Construct a Rust DateTime from an existing QDateTime, this is a copy operation.
+    pub fn from_qdatetime(qdatetime: &QDateTime) -> Self {
+        Self {
+            // Safety: TODO
+            inner: unsafe {
+                let mut ptr = MaybeUninit::<cxx::UniquePtr<QDateTime>>::zeroed();
+                qdatetime_init_from_qdatetime(&mut ptr, qdatetime);
+                ptr.assume_init()
+            },
+        }
+    }
+
+    /// Construct a Rust DateTime from a given QDate and QTime
+    pub fn from_date_time(date: &QDate, time: &QTime) -> Self {
```

Comment:
Maybe we should rename this to `from_date_and_time`.

I expected it to actually take a DateTime at first glance.

Reply:
Done